soc/nordic/nrf54h/soc.c

```diff
@@ -52,13 +60,17 @@ static void power_domain_init(void)
 	 *  WFI the power domain will be correctly retained.
 	 */
 
-	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0,
-			!IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD));
-	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_MAIN,
-			!IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD));
-
+#if !defined(CONFIG_SOC_NRF54H20_CPURAD)
+#if !(defined(CONFIG_PM) || defined(CONFIG_POWEROFF))
+	soc_lrcconf_poweron_request(&soc_node, NRF_LRCCONF_POWER_DOMAIN_0);
+#endif /* !(defined(CONFIG_PM) || defined(CONFIG_POWEROFF)) */
```

Comment:
This should apply to the Radio CPU as well to make sure the cache is properly handled in there.

Reply:
According to my understanding, you don't need to enable retention in radio lrcconfs. However, you need to force the domain on to keep cache powered. If not, the cace would lose its state when cpu gets retained